### ISSUE TITLE
regression: Kept mutations must have known mutation times.

### ISSUE THREAD

**Issue by grahamgower** [tskit-dev/msprime#1302]
Calling `msprime.mutate(..., keep=True)` on a tree sequence without mutation times raises an error (e.g. below with tree sequence from SLiM 3.4). This worked fine with msprime 0.7.4. Found when testing stdpopsim against msprime HEAD: see https://github.com/popsim-consortium/stdpopsim/pull/689#issuecomment-725916819.
```
ERROR: test_recap_and_rescale (tests.test_slim_engine.TestAPI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/stdpopsim/stdpopsim/tests/test_slim_engine.py", line 161, in test_recap_and_rescale
    ts1 = engine.simulate(
  File "/home/runner/work/stdpopsim/stdpopsim/stdpopsim/slim_engine.py", line 1123, in simulate
    ts = self._recap_and_rescale(
  File "/home/runner/work/stdpopsim/stdpopsim/stdpopsim/slim_engine.py", line 1248, in _recap_and_rescale
    msprime.mutate(
  File "/usr/share/miniconda/envs/stdpopsim/lib/python3.9/site-packages/msprime/mutations.py", line 1191, in mutate
    _msprime.sim_mutations(
_msprime.LibraryError: Kept mutations must have known mutation times.
```

**Comment by jeromekelleher** [comment]
Hmm - what should we do here @petrelharp?

**Comment by petrelharp** [comment]
Well, SLiM 3.5 will produce mutation times, but people will have old tree sequences lying around. The work-around is to run [compute_mutation_times](https://tskit.readthedocs.io/en/latest/python-api.html#tskit.TableCollection.compute_mutation_times) on the tree sequence before mutating. I guess we should modify the error message to direct people to this? How about
```
_msprime.LibraryError: Kept mutations must have known mutation times (which can be added using compute_mutation_times).
```
You may need to also pass `kept_mutations_before_end_time=True` here, another confusing thing we'll need to clear up.

**Comment by mufernando** [comment]
yes, I was compunting mutation times in my [PR ](https://github.com/popsim-consortium/stdpopsim/pull/644/files#diff-971e0c3ab37a1dafcaa06c7ff25e52b2fd5803b10a2f1cec3a469bcd903c7e7dR1204).

also opened a new [issue ](https://github.com/popsim-consortium/stdpopsim/issues/698)in stdpopsim to fix the kept mutations

**Comment by petrelharp** [comment]
I think this was closed in #1315, re-open if the solution isn't satisfactory.
